mux.go

```diff
@@ -107,6 +109,12 @@ func (mx *Mux) Options(pattern string, handlers ...interface{}) {
 	mx.handle(mOPTIONS, pattern, handlers...)
 }
 
+// NotFoundHandler sets a custom handler for the case when no routes matched
+// the given URL
+func (mx *Mux) NotFoundHandler(handler http.HandlerFunc) {
```

Comment:
I could make it to be just interface{} but I'm concerned about providing context for URL that will be missing which can be confusing. 

Also, if NotFound is made as middleware do you have idea how it will know about non existing addresses for middleware chained after it? 

Alternatively, I see NotFound handler to be a part of context. In such case NotFound middleware can change it from some default value.